app/models/katello/concerns/smart_proxy_extensions.rb

```diff
@@ -25,6 +25,12 @@ module SmartProxyExtensions
         before_create :associate_lifecycle_environments
         attr_accessible :lifecycle_environment_ids
 
+        has_many :containers,
+                 :class_name => "Container",
+                 :foreign_key => :capsule_id,
+                 :inverse_of => :capsule,
+                 :dependent => :nullify
```

Comment:
Nullify Containers that were provisioned using this smart proxy when the Smart Proxy is deleted? I think as long as the Compute Resource sees them, it's good. Removing the Smart Proxy just would impede provisioning a new container using it, but it shouldn't do anything to existing ones.